cluster/addons.go

```diff
@@ -468,7 +468,8 @@ func (c *Cluster) deployWithKubectl(ctx context.Context, addonYaml string) error
 }
 
 func (c *Cluster) doAddonDeploy(ctx context.Context, addonYaml, resourceName string, isCritical bool) error {
-	if c.UseKubectlDeploy {
+	if c.UseKubectlDeploy || (resourceName == NetworkPluginResourceName && !*c.Services.Kubeproxy.Enabled) {
+		// deploy via local kubectl if specified or if this is a network plugin and kubeproxy is disabled
```

Comment:
This describes whats happening but not why, why the exception on this case

Reply:
I updated the comment.
IIRC it's because the in-cluster addon deploy jobs won't work if you don't have kube-proxy as the basic networking component active (or a replacement for it).